tests/system/environmentLink_test.go

```diff
@@ -0,0 +1,36 @@
+package system
+
+import (
+	"github.com/quintilesims/layer0/tests/system/clients"
+	"testing"
+	"time"
+)
+
+// Test Resources:
+// This test creates two linked environments named 'el_public' and 'el_private`
+// The 'el_public' environment has a STS service running behind a public load balancer
+// The 'el_private' environment has a STS service running behind a private load balancer
+func TestEnvironmentLink(t *testing.T) {
```

Comment:
[nit] consider adding an additional unlinked environment that attempts to connect to internal resources in either environment, or some variation of this idea